### PR TITLE
t/85: Linking an image should not be possible

### LINKED ISSUE #85: You shouldn't be able to link an image (just yet)
The link feature should be disabled when an image is selected. Most likely some state check is missing.

In the future, we'll add the possibility to link images, but so far we don't have converters for that.

Alternatively, instead of fixing the state check now, we can add a converter allowing linking images. But the state check needs to be fixed at some point anyway.

### PR DESCRIPTION
**Pull request by oleq**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Linking an entire image should not be possible. Closes #85.

### CODE REVIEW

File: tests/integration.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global document */
+
+import Link from '../src/link';
+import Image from '@ckeditor/ckeditor5-image/src/image';
```
Comment (oskarwrobel) [inline tests/integration.js]: Image should be added as a devDependency.

**Comment by Reinmar** [comment]
This change is bad because it couples image and link features. This should not happen this way. This should be managed through schema and `checkAttributeInSelection()`. So, unless there's some problem to control this through schema (in which case it must be mentioned in https://github.com/ckeditor/ckeditor5-engine/issues/532) this PR should be reverted.

**Comment by oleq** [comment]
When the `ImageCaption` is on, linking doesn't work because `doc.schema.checkAttributeInSelection( doc.selection, 'linkHref' )` indeed returns `false` in a such  situation like `[<image></image>]`.

However, when `ImageCaption` is on, the following rules are added to the schema:

```js
schema.registerItem( 'caption', '$block' );
schema.allow( { name: '$inline', inside: 'caption' } );
schema.allow( { name: 'caption', inside: 'image' } );
```

which make `checkAttributeInSelection` return `true` when
```
[<image><caption>a</caption></image>]
```
and `false` when
```
[<image><caption></caption></image>]
```
When there's a text in `<caption>`, it automatically means the linking is possible. I have no idea what to use to check if an attribute is allowed `<image>here<caption>...</caption></image>` instead of `<image><caption>there</caption></image>` for any selection. The first, shallow check would certainly solve the issue.

**Comment by Reinmar** [comment]
OK, I know what's wrong here, actually. It's `checkAttributeInSelection()`. And the issue is strongly related to https://github.com/ckeditor/ckeditor5-engine/issues/986. The same kind of fix is needed in `checkAttributeInSelection()`, but... this will disable not only link but also bold and italic. So, I'm not sure if we want to do that.

Let's revert this change and get back to the original ticket.

**Pull request merged**